apps/condo/domains/ticket/hooks/usePollTicketComments.tsx

```diff
@@ -0,0 +1,70 @@
+import { SortTicketCommentsBy } from '@app/condo/schema'
+import get from 'lodash/get'
+import uniq from 'lodash/uniq'
+import { useCallback, useEffect, useRef } from 'react'
+
+import { useBroadcastChannel } from '@condo/domains/common/hooks/useBroadcastChannel'
+import { useExecuteWithLock } from '@condo/domains/common/hooks/useExecuteWithLock'
+import { TicketComment } from '@condo/domains/ticket/utils/clientSchema'
+
+
+const COMMENT_RE_FETCH_INTERVAL_IN_MS = 5 * 1000
+const LOCK_NAME = 'ticketComments'
+const BROADCAST_CHANNEL_NAME = 'ticketComments'
+const LOCAL_STORAGE_SYNC_KEY = 'syncTicketCommentsAt'
+
+export function usePollTicketComments ({
+    ticket,
+    refetchTicketComments,
+}) {
+    const intervalRef = useRef<ReturnType<typeof setInterval> | null>()
+
+    const { refetch: refetchSyncComments } = TicketComment.useObjects({}, { skip: true })
+
+    const { sendMessage } = useBroadcastChannel<string[]>(BROADCAST_CHANNEL_NAME, async (ticketIdsWithUpdatedComments) => {
+        if (ticketIdsWithUpdatedComments.includes(get(ticket, 'id'))) {
+            await refetchTicketComments()
+        }
+    })
+
+    const pollTicketComments = useCallback(async () => {
+        if (!localStorage) return
```

Comment:
what are the reasons for this check?

Reply:
for check that it's client side and localStorage is exist